src/helpers/getPSP22Balances.ts

```diff
@@ -116,35 +119,37 @@ export const watchPSP22Balances = (
   const fetchTokenBalances = async () =>
     callback(
       await Promise.all(
-        Object.values(allPSP22Assets).map(async ({ slug, decimals, symbol, iconPath }) => {
-          let balance = new BN(0)
-
-          const contract = psp22ContractMap[slug]
-          const _balanceOf = await contract.query['psp22::balanceOf'](
-            address,
-            { gasLimit: getMaxGasLimit(api) },
-            address,
-          )
-          const balanceObj = _balanceOf?.output?.toPrimitive() as Record<string, any>
-
-          balance = new BN(
-            _balanceOf.output ? (balanceObj.ok as string) || (balanceObj.Ok as string) : '0',
-          )
-
-          const data = {
-            tokenDecimals: decimals,
-            tokenSymbol: symbol,
-            balance,
-          }
-
-          const balanceFormatted = parsePSP22Balance(data, formatterOptions)
-          return {
-            balanceFormatted,
-            tokenSlug: slug,
-            iconPath,
-            ...data,
-          }
-        }),
+        Object.values(allPSP22Assets)
+          .filter(({ originChain }) => originChain === activeChain)
+          .map(async ({ slug, decimals, symbol, iconPath }) => {
+            let balance = new BN(0)
+
+            const contract = psp22ContractMap[slug]
+            const _balanceOf = await contract.query['psp22::balanceOf'](
```

Comment:
Why are you not using our `contractQuery` helper?

Reply:
Same as above.